examples/hotrod/services/driver/client.go

```diff
@@ -30,25 +30,23 @@ import (
 
 // Client is a remote client that implements driver.Interface
 type Client struct {
-	tracer opentracing.Tracer
 	logger log.Factory
 	client DriverServiceClient
 }
 
 // NewClient creates a new driver.Client
-func NewClient(tracer opentracing.Tracer, logger log.Factory, hostPort string) *Client {
+func NewClient(tracerProvider trace.TracerProvider, logger log.Factory, hostPort string) *Client {
 	conn, err := grpc.Dial(hostPort, grpc.WithTransportCredentials(insecure.NewCredentials()),
 		grpc.WithUnaryInterceptor(
-			otgrpc.OpenTracingClientInterceptor(tracer)),
+			otelgrpc.UnaryClientInterceptor(otelgrpc.WithTracerProvider(tracerProvider))),
 		grpc.WithStreamInterceptor(
-			otgrpc.OpenTracingStreamClientInterceptor(tracer)))
+			otelgrpc.StreamClientInterceptor(otelgrpc.WithTracerProvider(tracerProvider))))
```

Comment:
Are we still planning to split this PR into more pieces? I prefer to have PRs that deal with one problem at a time, and we have the baggage issue that might be more complicated across several services, so it's best to merge all the OTEL upgrades before tackling it.

Reply:
I see. Let me grep it over and give you update here.

Reply:
Sorry Yuri, I am not able to split this PR. We are getting stuck while considering `NewServer` creation.

Reply:
that's probably because you're introducing too many changes in the library functions, which I don't think are necessary.